src/lib.rs

```diff
@@ -70,15 +70,29 @@ where
 fn sanitize_path(uri: &mut Uri) {
     let path = uri.path();
     let path_decoded = decode(path);
+
+    // Check if the path contains a trailing slash.
+    let trailing_slash = if let Some(char) = path_decoded.chars().last() {
+        char == '/'
+    } else {
+        false
+    };
```

Comment:
This could be greatly simplified with `is_some_and(|c| c == '/')`, but it would bump the msrv of this crate to 1.70 (from <1.63).